go/config/config.go

```
@@ -276,6 +283,8 @@ func (config *ClientConfiguration) WithReadFrom(readFrom ReadFrom) *ClientConfig
 // encompasses sending the request, awaiting for a response from the server, and any required reconnections or retries. If the
 // specified timeout is exceeded for a pending request, it will result in a timeout error. If not set, a default value will be
 // used.
+//
+// WARNING: If requestTimeout exceeds the max duration of 2^32 - 1 milliseconds, a default value will be used.
```

Should also mention that this is the case for negative durations.

---

Also, this maximum value is [equal to over 49 days](https://go.dev/play/p/Jyh50h5QgRW). Perhaps if someone passes 50 days in the code should just panic instead of ignoring the bad value silently.

---

I agree. I think instead of doing a panic, we can throw an error instead. I've changed the function from returning a bool to an error instead and kept the 0 vs not 0 case in toProtobuf to account for this change